### PR TITLE
[BugFix] Fix convert json object with empty key

### LINKED ISSUE #16851: Parse json with empty key failed
### Steps to reproduce the behavior (Required)
Load json data :
```json
{
  "a": {
      "": 1
  }
}
```

### Real behavior (Required)
```c++
terminate called after throwing an instance of 'arangodb::velocypack::Exception'
  what():  The key of the next key/value pair must be a string
query_id:00000000-0000-0000-0000-000000000000, fragment_instance:00000000-0000-0000-0000-000000000000
*** Aborted at 1674552585 (unix time) try "date -d @1674552585" if you are using GNU date ***
PC: @     0x7f1d77f66387 __GI_raise
*** SIGABRT (@0x6eee) received by PID 28398 (TID 0x7f1d557ca700) from PID 28398; stack trace: ***
    @          0x59fe902 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7f1d78a1b630 (unknown)
    @     0x7f1d77f66387 __GI_raise
    @     0x7f1d77f67a78 __GI_abort
    @          0x2b00334 _ZN9__gnu_cxx27__verbose_terminate_handlerEv.cold
    @          0x7d22ef6 __cxxabiv1::__terminate()
    @          0x7d22f61 std::terminate()
    @          0x7d23106 __cxa_rethrow
    @          0x2900ddd _ZN9starrocks17SimdJsonConverter7convertEN8simdjson8fallback8ondemand5valueESt17basic_string_viewIcSt11char_traitsIcEEPN8arangodb10velocypack7BuilderE.isra.0.cold
    @          0x4804e16 starrocks::SimdJsonConverter::convert()
    @          0x4805542 starrocks::SimdJsonConverter::convert()
    @          0x4806033 starrocks::SimdJsonConverter::create()
    @          0x4805f3b starrocks::convert_from_simdjson()
    @          0x4802c4a starrocks::JsonValue::from_simdjson()
    @          0x3edb33e starrocks::vectorized::add_native_json_column()
    @          0x3eca73a starrocks::vectorized::add_nullable_column()
    @          0x3ecb423 starrocks::vectorized::add_nullable_column()
    @          0x2eaf0e5 starrocks::vectorized::JsonReader::_construct_row_without_jsonpath()
    @          0x2eb460f starrocks::vectorized::JsonReader::_read_rows<>()
    @          0x2eafdd9 starrocks::vectorized::JsonReader::read_chunk()
    @          0x2eb001c starrocks::vectorized::JsonScanner::get_next()
    @          0x4dc7781 starrocks::connector::FileDataSource::get_next()
    @          0x2facf55 starrocks::vectorized::ConnectorScanNode::_scanner_thread()
    @          0x46c9600 starrocks::PriorityThreadPool::work_thread()
    @          0x59be307 thread_proxy
    @     0x7f1d78a13ea5 start_thread
    @     0x7f1d7802eb0d __clone
    @                0x0 (unknown)
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - 2.2, 2.3, 2.4, 2.5

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #16851

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [x] 2.2

### CODE REVIEW

File: be/src/util/json_converter.cpp [hunk 1]
```diff
@@ -56,116 +56,118 @@ class SimdJsonConverter {
...
             builder->add(toStringRef(field_name), vpack::Value(value));
         } else {
             builder->add(vpack::Value(value));
         }
         return Status::OK();
     }
 
-    static inline Status convert_null(std::string_view field_name, vpack::Builder* builder) {
-        if (!field_name.empty()) {
+    static inline Status convert_null(std::string_view field_name, bool has_name, vpack::Builder* builder) {
```
Comment (chaoyli) [inline be/src/util/json_converter.cpp:169]: using `bool has_name=false` may be better.

File: be/src/util/json_converter.cpp [hunk 2]
```diff
@@ -56,116 +56,118 @@ class SimdJsonConverter {
...
+        if (has_name) {
             builder->add(toStringRef(field_name), vpack::Value(vpack::ValueType::Object));
         } else {
             builder->add(vpack::Value(vpack::ValueType::Object));
         }
         for (auto field : obj) {
             std::string_view key = field.unescaped_key();
             auto value = field.value().value();
-            RETURN_IF_ERROR(convert(value, key, builder));
+            RETURN_IF_ERROR(convert(value, key, true, builder));
```
Comment (chaoyli) [inline be/src/util/json_converter.cpp:98]: How to determine passing true in this place?

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

File: be/src/util/json_converter.cpp (hunk 1 above)
Comment (mofeiatwork) [reply be/src/util/json_converter.cpp:169]: renamed to `is_object`

File: be/src/util/json_converter.cpp (hunk 2 above)
Comment (mofeiatwork) [reply be/src/util/json_converter.cpp:98]: - true means object field, which has a field name
- false means array

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**